tests/webTables.spec.ts

```diff
@@ -22,22 +20,24 @@ test.describe('Web Tables', () => {
         await expect(page.getByRole('row', { name: 'Madison' })).toHaveCount(4)
     })
 
-
-
     test('Test Case 3: Validate search by Last Name', async ({ page }) => {
         await page.getByRole('button', { name: 'Owners' }).click()
         await page.getByRole('link', { name: 'Search' }).click()
-        await page.locator('#lastName').fill('Black')
+        const ownersLastName = page.locator('#lastName')
+        await ownersLastName.fill('Black')
+        await page.getByRole('button', { name: 'Find Owner' }).click()
+        await expect(page.getByRole('row', { name: '1450 Oak Blvd.' }).locator('td').nth(0)).toContainText('Black')
+        await ownersLastName.clear()
+        await ownersLastName.fill('Davis')
         await page.getByRole('button', { name: 'Find Owner' }).click()
-        await expect(page.locator('tbody').getByRole('cell', { name: 'Jeff Black' })).toContainText('Black')
-        await page.locator('#lastName').clear()
-        await page.locator('#lastName').fill('Davis')
-        await expect(page.locator('tbody').getByRole('cell', { name: 'Betty Davis' })).toContainText('Davis')
+        await expect(page.locator('td').nth(0)).toContainText('Davis')
```

Comment:
The test is no longer a "false positive," but it still does not match the requirements of the test case.

The test step says: "Add the assertion that each owner displayed in the table has a last name "Davis"".
Your script only validates that the first column of the first row of the table contains "Davis", but instead, you need to validate that every row of the table has the same value in the first column.

Also, you have changed the approach in locator selection between two owners. Why?
For the owner "Black" you have used the address to find the row of the table and index but for the "Davis" you have used just the index. You need to be consistent with the locator selection.
Technically, you don't need address of the owner to get the rows of the table with owners.

Also, look closely at the remaining of the steps. After you have validated "Black" and "Davis" you begin a loop with the other two values "ES" and "Playwright" to continue the validation with the same goal. So for optimization, you can add "Davis" and "Black" to the same loop, so a single loop will perform a validation of all 4 values instead of 2. This is the same way we did in the lesson. You can check the lesson again for the reference.

If something is not clear and you need additional guidance, please let me know